web/src/components/molecules/Schema/FieldModal/FieldCreationModal/index.tsx

```diff
@@ -0,0 +1,245 @@
+import styled from "@emotion/styled";
+import React, { useCallback } from "react";
+
+import Button from "@reearth-cms/components/atoms/Button";
+import Checkbox from "@reearth-cms/components/atoms/Checkbox";
+import Form from "@reearth-cms/components/atoms/Form";
+import Icon from "@reearth-cms/components/atoms/Icon";
+import Input from "@reearth-cms/components/atoms/Input";
+import Modal from "@reearth-cms/components/atoms/Modal";
+import Tabs from "@reearth-cms/components/atoms/Tabs";
+import TextArea from "@reearth-cms/components/atoms/TextArea";
+import FieldDefaultProps from "@reearth-cms/components/molecules/Schema/FieldModal/FieldDefaultProps";
+import FieldValidationProps from "@reearth-cms/components/molecules/Schema/FieldModal/FieldValidationProps";
+import { SchemaFieldTypePropertyInput, SchemaFiledType } from "@reearth-cms/gql/graphql-client-api";
+
+import { fieldTypes } from "../../fieldTypes";
+import { FieldType } from "../../types";
+
+export interface FormValues {
+  title: string;
+  description: string;
+  key: string;
+  multiValue: boolean;
+  unique: boolean;
+  required: boolean;
+  type: SchemaFiledType;
+  typeProperty: SchemaFieldTypePropertyInput;
+}
+
+export interface Props {
+  open?: boolean;
+  selectedType: FieldType | null;
+  onClose?: (refetch?: boolean) => void;
+  onSubmit?: (values: FormValues) => Promise<void> | void;
+}
+
+const initialValues: FormValues = {
+  title: "",
+  description: "",
+  key: "",
+  multiValue: false,
+  unique: false,
+  required: false,
+  type: SchemaFiledType["Text"],
+  typeProperty: { text: { defaultValue: "", maxLength: 0 } },
+};
+
+const FieldCreationModal: React.FC<Props> = ({ open, onClose, onSubmit, selectedType }) => {
+  const [form] = Form.useForm();
+  const { TabPane } = Tabs;
+
+  const handleSubmit = useCallback(() => {
+    form
+      .validateFields()
+      .then(async values => {
+        values.type = selectedType;
+        if (selectedType === "Text") {
+          values.typeProperty = {
+            text: { defaultValue: values.defaultValue, maxLength: values.maxLength },
+          };
+        }
+        if (selectedType === "TextArea") {
+          values.typeProperty = {
+            textArea: { defaultValue: values.defaultValue, maxLength: values.maxLength },
+          };
+        }
+        if (selectedType === "MarkdownText") {
+          values.typeProperty = {
+            markdownText: { defaultValue: values.defaultValue, maxLength: values.maxLength },
```

Comment:
I thought we discussed this and agreed to NOT do a list of if statements because its harder for the developer. Please choose either else/if or a switch statement and update here.

Reply:
Also, what happens if selectedType is Null? Is it okay to be able to submit with the default value for typeProperty? I feel like that shouldn't be allowed.

Reply:
@KaWaite , the only way to have it `null` is if it's not chosen yet... 
There is no case where the modal is opened and the selected type is null, if so the whole modal won't work properly for several reasons.

Reply:
The type of selectedType includes null. If it will never be null please update the type

Reply:
Also, what about the else/if or switch?